### PR TITLE
Fix suboptimal numeric perturbation for LIME

### PR DESCRIPTION
**Pull request by tteofili**
Many thanks for submitting your Pull Request :heart:!

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](https://github.com/trustyai-explainability/trustyai-explainability/blob/main/CONTRIBUTING.md)
- [x] Pull Request title is properly formatted: `FAI-XYZ Subject`
- [x] Pull Request title contains the target branch if not targeting main: `[0.3.x] FAI-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

### CODE REVIEW

File: explainability-core/src/main/java/org/kie/trustyai/explainability/model/PerturbationContext.java [hunk 1]
```diff
@@ -27,21 +27,34 @@
...
     private final Optional<Long> seed;
 
     private final Random random;
 
     private final int noOfPerturbations;
 
+    private final double standardDeviation;
+
     public PerturbationContext(Random random, int noOfPerturbations) {
+        this(random, noOfPerturbations, DEFAULT_STANDARD_DEVIATION); // preserving old behavior
```
Comment (danielezonca) [inline explainability-core/src/main/java/org/kie/trustyai/explainability/model/PerturbationContext.java:40]: Do we want to preserve the old behaviour? If we are changing this behaviour because suboptiman/broken I would probably change also this. I don't think it is an issue to change this

**Review by danielezonca — APPROVED** [review]

File: explainability-core/src/main/java/org/kie/trustyai/explainability/model/PerturbationContext.java (hunk 1 above)
Comment (tteofili) [reply explainability-core/src/main/java/org/kie/trustyai/explainability/model/PerturbationContext.java:40]: I think this can be safely changed, I just would like to get evidence from real benchmarks to choose the new value. I've adjusted the LIME retry mechanism to multiply this by 10 automatically on each retry, so it'll be 0.01 on the first try, 1 on the second, 10 on the third, etc.
tldr; I'd keep this until we get a better default from experiments with real data.
